### PR TITLE
Add windows build pipeline

### PR DESCRIPTION
**Pull request by paryoja**
### What is the context for this pull request?

 - **Tracking Issue**: #376  #469

### What changes were proposed in this pull request?

Using Windows environments to test in the azure pipeline.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

By using the Azure pipeline configured.

### CODE REVIEW

**Comment by paryoja** [comment]
I added four windows jobs according to linux jobs already exist.
However, windows jobs (~20mins) takes more time than linux's (~12min).
I think it could be a bit annoyed if you wait for your job to be completed.

**Comment by paryoja** [comment]
I think I should do some refactoring so that this pipeline could be easily used after new versions of Scala / Spark are released.

**Comment by paryoja** [comment]
Refactoring Done.

File: ci/windows_test.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      pwsh: true
+      targetType: inline
+      script: |
+        echo "Download Hadoop utils for Windows."
+        curl -k -L -o hadoop.zip https://github.com/steveloughran/winutils/releases/download/tag_2017-08-29-hadoop-2.8.1-native/hadoop-2.8.1.zip
+        unzip hadoop.zip
+        New-Item -ItemType Directory -Force -Path hadoop\bin
+        cp hadoop-2.8.1\winutils.exe hadoop\bin
+        cp hadoop-2.8.1\hadoop.dll hadoop\bin
+        cp hadoop-2.8.1\hadoop.dll c:\windows\system32
```
Comment (imback82) [inline ci/windows_test.yml:19]: oh, this is required?

**Review by imback82 — APPROVED** [review]
LGTM (one question), thanks @paryoja!

**Comment by imback82** [comment]
> I added four windows jobs according to linux jobs already exist.
> However, windows jobs (~20mins) takes more time than linux's (~12mins).
> I think it could be a bit annoyed if you wait for your job to be completed.

It should be OK for now, and we can enable Windows builds only for CI builds on master if the speed up is needed.

> I think I should do some refactoring so that this pipeline could be easily used after new versions of Scala / Spark are released.

Thanks for doing this!

File: ci/windows_test.yml (hunk 1 above)
Comment (paryoja) [reply ci/windows_test.yml:19]: Except for the spark 3.1 build, it is not necessary. (2.4, 3.0 tests success without copying it to system32 dir)

For the spark 3.1 build, it seems that sbt test doesn't reference hadoop/bin/hadoop.dll so it makes the tests fail.
I don't fully understand the reason. Maybe Spark 3.1 doesn't reference HADOOP_HOME to include dynamic libraries.

**Pull request merged**